deploy-site.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+echo -e "\033[0;32mDeploying website...\033[0m"
+
+cd site && npx hexo generate
+
+# delete old deploy branch
+git branch -D deploy
+
+git checkout -b deploy
+
+# Add changes to git.
+git add -f public
+
+# Commit changes.
+msg="chore(*): adding public folder `date`"
+if [ $# -eq 1 ]
+  then msg="$1"
+fi
+git commit -m "$msg"
+
+cd ../
+
+git subtree split -P site/public -b deploy-dist
+
+# Push source and build repos.
+git push -f embark-site deploy-dist:gh-pages
+git branch -D deploy-dist
+git checkout master
+git branch -D deploy
+
+
```

Comment:
I had to barely touch this script. Only thing was to move it to the root of our repo, otherwise we can't use `git subtree`.

Reply:
Can you explain?  Does it have to do with the current working directory and how the script is invoked?

Reply:
`git subtree` can only be invoked from the root of a git repository.

Reply:
What if the script is in `<root>/scripts/deploy-site.sh`, same `scripts/` dir where `release.js` is located?

Will it work to invoke it this way?

```
$ cd <root>
$ scripts/deploy-site.sh
```

The current working directory would be the root of the git repository even though the script is located in `scripts/`.

Reply:
This might work! Lemme try that.

Reply:
I'm sorry actually, the reason I moved the deploy-site.sh script in the root was because exactly that didn't work haha

Reply:
No problem, I wasn't sure if maybe you were invoking the script from the directory in which it was located instead of invoking it from the root.

Reply:
I stand corrected. @michaelsbradleyjr your guess was spot on. I was testing a different setting